### ISSUE TITLE
ERROR:hicexplorer.hicBuildMatrix:'KI270466.1'

### ISSUE THREAD

**Issue by snowlover** [maxplanck-ie/snakepipes#697]
Hi,

everytime the HiC pipeline stopped after this error in files:
`HiC_matrices/logs/*kb.err`

```
...
INFO:hicexplorer.hicBuildMatrix:13908187 (8.04%) valid pairs added to matrix

INFO:hicexplorer.hicBuildMatrix:processing 183094696 lines took 6195.20 secs (29554.3 lines per second)

INFO:hicexplorer.hicBuildMatrix:13908187 (7.60%) valid pairs added to matrix

ERROR:hicexplorer.hicBuildMatrix:'KI270466.1'
```

no other information was available for debugging. try setting 'chromosomes: chr1 .... chrX ' did not help.

any suggestions ?

also, will lowering resolution from 100kb (current setting) to 1Mb increase the valid pairs rate (8% here) ?

Thanks a lot.

**Comment by LeilyR** [comment]
which version are you using? also could you please send the command you were using and the related part of your log file to this error.

**Comment by snowlover** [comment]
version 2.2.3 and the pipeline command is
`HiC -i hic-fq -o hic-out --ext .fq.gz --local -j 10 --binSize 100000 --correctionMethod KR hg38`

I've posted the related part of my log file for building matrix. the following is from snakemake:

```
...
Error in rule build_matrix:
    jobid: 13
    output: HiC_matrices/g2_bs100kb.cool, HiC_matrices/QCplots/g2_QC/QC.log
    log: HiC_matrices/logs/g2_bs100kb.out, HiC_matrices/logs/g2_bs100kb.err (check log file(s) for error message)
    conda-env: /home/user/miniconda3/envs/4c5483f1
    shell:
        hicBuildMatrix -s BWA/g2_R1.bam BWA/g2_R2.bam -bs 100000 -rs DpnII.bed --restrictionSequence GATC --danglingSequence GATC --minDistance 150 --maxDistance 1000 --QCfolder HiC_matrices/QCplots/g2_QC/ --threads 10  -o HiC_matrices/g2_bs100kb.cool > HiC_matrices/logs/g2_bs100kb.out 2> HiC_matrices/logs/g2_bs100kb.err && rm HiC_matrices/QCplots/g2_QC/QC_table.txt
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)
...
```

if you need more information, please let me know.
Thanks.

**Comment by joachimwolff** [comment]
HiCExplorer 2.2.3? Please update to version 3.5!

**Comment by snowlover** [comment]
no, I mean snakePipes 2.2.3 (latest), HiCexplorer is 3.5 already

**Comment by LeilyR** [comment]
Hi @snowlover , the command line seems to be fine to me. It looks as if there is something wrong with the hicexplorer in handling a contig that you have. But just be certain that issue is not coming from the pipeline itself, would you mind running the shell command you have sent independently (after activating its corresponding conda env) and  send as the full error message (both .out and .err). If this is a bug in hicexplorer I will change the version in our pipeline which can use it with no crash and transfer your issue to hicexplorer repository.

**Comment by snowlover** [comment]
thanks for your help, but this is a totally new run, so I think no information other than those provided here I can get from a re-run. It looks like the extra contigs results in the error. Maybe the bam file need to remove those contigs ?

**Comment by LeilyR** [comment]
sorry I might have been confused with another issue which has been opened today where the error was coming from `Bio.Alphabet` . I think the only thing left in your case would be trying to run the command line outside the snakepipes which I am almost certain that will return the same error. I am quite convinced that it is an issue coming from hicexplorer and how it handles contigs of strange names. I will open an issue on hicexplorer. Meanwhile maybe you could get rid of the strange contigs if possible and try it without them. I hope it helps.

**Comment by snowlover** [comment]
yeah, thanks.

**Comment by snowlover** [comment]
I still have the question of valid pair rate (~8% here), so I reopened this issue, and maybe wait for the hicexplorer issue solution

**Comment by LeilyR** [comment]
yeah 8% is way too low , I agree, as if the samples failed. Do you want to send a pair fastq files of you and I will have a look.

**Comment by snowlover** [comment]
right, I'm worried ... sure, how many reads should I include ?

**Comment by LeilyR** [comment]
no it is not if you have no read to begin with.

**Comment by LeilyR** [comment]
Hi @snowlover , 

I have tested the hic pipeline on a data set with several contigs and could reproduce the same issue as you had. This is related to a bug which has been introduced in hicexplorer version 3.5. The work around for now could be using an older version of snakepipes. I have used snakepipes/2.0.0 which runs on hicexplorer/3.4 and it solved the problem.

**Comment by LeilyR** [comment]
On top of that about the contig names failure , please use snakepipes/2.0.0 which runs on hicexplorer/3.4 this will solve the contig name issue.

**Comment by joachimwolff** [comment]
Fixed with HiCExplorer 3.5.3 release..

**Comment by katsikora** [comment]
this is now available in release 2.3.1.
